### ISSUE TITLE
Blood pressures should return as integers rather than floats

### ISSUE THREAD

**Issue by shabiel** [synthetichealth/synthea#555]
I have not seen any sphygmomanometer that can produce blood pressures such as these:

```
jq '.entry[].resource | select(.code.coding) | select (.code.coding[0].code == "55284-4") | { sys: .component[0].valueQuantity.value, dia: .component[1].valueQuantity.value } ' < output/fhir/Alfonso758_Gálvez271_0270ece3-d93e-49e5-abfa-44d5cf30e988.json 
{
  "sys": 88.29144415948353,
  "dia": 135.23018803524187
}
{
  "sys": 85.38462369227672,
  "dia": 113.6810154467861
}
{
  "sys": 81.13054817674676,
  "dia": 114.95844871239585
}
{
  "sys": 83.45264439371225,
  "dia": 118.82586622254128
}
{
  "sys": 72.77896457826508,
  "dia": 121.73250910823609
}
{
  "sys": 85.14656872532069,
  "dia": 123.21180424227781
}
{
  "sys": 81.30910581978247,
  "dia": 118.51571585691588
}
{
  "sys": 75.86713841285172,
  "dia": 134.6475547640152
}
{
  "sys": 77.74549782225387,
  "dia": 101.45816752987017
}
{
  "sys": 84.91786256626268,
  "dia": 128.25111524136327
}
```

A realistic blood pressure would be something like 128/84 with no decimal point.

I would like to help fix this, if I know where to look. Looks like `src/main/java/org/mitre/synthea/modules/BloodPressureValueGenerator.java` is the right file to look in.

**Comment by jawalonoski** [comment]
I think it is probably the exporters you want to modify, and have them output integers (or truncate the decimals to a few places), rather than messing with the internal data representation.

**Comment by shabiel** [comment]
That unfortunately means you have to change 6 files (maybe more) rather than one. Are you sure?

As far as internal data--I agree we need to retain the decimals until we have the final blood pressure.

**Comment by jawalonoski** [comment]
Well, the alternative is to put special logic into `State.Observation.process(...)` for different types of vital signs and convert to integer when a `new HealthRecord.Observation(...)` is created.

So, add some `if` or `switch` logic inside the block on line 1174 to convert to an `Integer` (it has to be an Object type and not a primitive) when it is a Blood Pressure reading.

The issue with this approach is that we would be putting special logic into a relatively generic class.

https://github.com/synthetichealth/synthea/blob/9ab5bebc11fc159b6b260fff07cc638025d35de6/src/main/java/org/mitre/synthea/engine/State.java#L1173-L1178

**Comment by shabiel** [comment]
Thank you. I will take a look.

**Comment by shabiel** [comment]
I took a quick look at the debugger. Thank you for your help.

It looks like the issue applies to all vital signs. I have a pain score of 1.0668997618663942, which doesn't make sense.

My current thought process is that each vital will need to keep its internal representation, but each will need to provide a custom rounded out version of itself (integers for blood pressures, pain scores, one decimal place for height and weight). What do you think?

**Comment by jawalonoski** [comment]
Sounds okay in theory, but the devil is in the details...

Here are my stream of thoughts...

Where are you going to track this? 
 - In the `Person` with `Person.getVitalSign(...)`?
 - In the `ValueGenerator` classes?
 - In the `HealthRecord` with `HealthRecord.Observation` constructor?
 - In the `State.Observation` class?
 - In the `VitalSign` enum?

Putting that aside for a moment, say you manage to convert some values to integers, for example, blood pressures and pain scores... for the others, how do you propose to manage the decimal places in the JSON?

I think you can try something with `BigDecimal.setScale(...)` but I'm not sure what the HAPI FHIR libraries will end up doing with it once it starts converting to JSON... never mind the other export formats...
https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#setScale-int-java.math.RoundingMode- 

You might want to quickly hack the FHIR R4 Observation exporter with `BigDecimal.setScale(...)`, and see what happens, before you invest too much time into it. If that works, you can move it upstream somewhere that makes sense.

https://github.com/synthetichealth/synthea/blob/8623c1af899b059584760e63aded810b854ea282/src/main/java/org/mitre/synthea/export/FhirR4.java#L1405-L1408

**Comment by shabiel** [comment]
Thank you for your guidance. I will see what I can do. I was hoping
for an "easy" fix, for a simple contribution to the source code to get
me started... oh well.

**Comment by jawalonoski** [comment]
Don't be discouraged! Go for it!

**Comment by shabiel** [comment]
It seems to work fine! How surprising. Didn't crash, and the FHIR R4 classes seemed to export fine.

Code
```
return new Quantity().setValue(BigDecimal.valueOf((Double)value).setScale(1, BigDecimal.ROUND_HALF_UP))
```

Output
```
        "valueQuantity": {
          "value": 163.1,
          "unit": "cm",
          "system": "http://unitsofmeasure.org",
          "code": "cm"
        }
```

**Comment by jawalonoski** [comment]
Fixed with #560 and #573
